src/wallet/Wallet.ts

```diff
@@ -0,0 +1,2192 @@
+	 * @param p2pkOptions P2PK locking options (e.g. pubkey, locktime).
+	 * @param config Optional parameters (e.g. includeFees).
+	 * @returns SendResponse with keep/send proofs.
+	 */
+	async sendAsP2PK(
+		amount: number,
+		proofs: Proof[],
+		p2pkOptions: P2PKOptions,
+		counter?: number,
+		config?: SendConfig,
+	): Promise<SendResponse> {
+		const keepOutput: OutputType = counter ? { type: 'deterministic', counter } : DEFAULT_OUTPUT;
+		return this.send(
+			amount,
+			proofs,
+			{ send: { type: 'p2pk', options: p2pkOptions }, keep: keepOutput },
+			config,
+		);
+	}
+
+	/**
+	 * Sends proofs with P2PK-locked change (keep) outputs (random for send).
+	 *
+	 * @remarks
+	 * For apps storing locked proofs. Uses NUT-11 options to lock change.
+	 * @param amount Amount to send.
+	 * @param proofs Proofs to split (sum >= amount).
+	 * @param p2pkOptions P2PK locking options for keep.
+	 * @param config Optional parameters (e.g. includeFees).
+	 * @returns SendResponse with keep/send proofs.
+	 */
+	async sendWithP2PKChange(
+		amount: number,
+		proofs: Proof[],
+		p2pkOptions: P2PKOptions,
+		config?: SendConfig,
+	): Promise<SendResponse> {
+		return this.send(
+			amount,
+			proofs,
+			{ send: DEFAULT_OUTPUT, keep: { type: 'p2pk', options: p2pkOptions } },
+			config,
+		);
+	}
+
+	/**
+	 * Splits and creates sendable tokens.
+	 *
+	 * @remarks
+	 * This method performs an online swap if necessary. The `outputConfig` defaults to
+	 * `DEFAULT_OUTPUT_CONFIG`, which uses random blinding factors for both `send` and `keep` outputs.
+	 * For common cases, use `sendAs...` helpers (eg sendAsDefault, sendAsP2PK etc). If proofs are
+	 * P2PK-locked to your public key, call signP2PKProofs first to sign them.
+	 * @example
+	 *
+	 * ```typescript
+	 * // Simple send (uses DEFAULT_OUTPUT_CONFIG)
+	 * const result = await wallet.send(5, proofs);
+	 *
+	 * // Use default output configuration
+	 * const result = await wallet.send(5, proofs, undefined, { includeFees: true });
+	 *
+	 * // Or explicitly use DEFAULT_OUTPUT_CONFIG
+	 * const result = await wallet.send(5, proofs, DEFAULT_OUTPUT_CONFIG, { includeFees: true });
+	 *
+	 * // Custom output configuration
+	 * const customConfig: OutputConfig = {
+	 * 	send: { type: 'p2pk', options: { pubkey: '...' } },
+	 * 	keep: { type: 'deterministic', counter: 0 },
+	 * };
+	 * const customResult = await wallet.send(5, proofs, customConfig);
+	 * ```
+	 *
+	 * @param amount Amount to send (receiver gets this net amount).
+	 * @param proofs Array of proofs to split.
+	 * @param outputConfig Configuration for send and keep (change) outputs.
+	 * @param config Optional parameters for the swap.
+	 * @returns SendResponse with keep/send proofs.
+	 * @throws Throws if the send cannot be completed offline or if funds are insufficient.
+	 */
+	async send(
+		amount: number,
+		proofs: Proof[],
+		outputConfig: OutputConfig = DEFAULT_OUTPUT_CONFIG,
+		config?: SendConfig,
+	): Promise<SendResponse> {
```

Comment:
In v3-alpha-2, I've reversed the 3rd and 4th params in send/receive/mint/melt - it makes for less of a breaking change upgrading from v2, and this removes the need for the DEFAULT_OUTPUT etc. if you want to apply a config.

So it is currently a breaking change to the signature in thie PR that goes away in next version